sdk/iot/hub/samples/telemetry/src/iot_hub_telemetry_example.c

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifdef _MSC_VER
+// warning C4201: nonstandard extension used: nameless struct/union
+#pragma warning(disable : 4201)
+#endif
+#include <paho-mqtt/MQTTClient.h>
+#ifdef _MSC_VER
+#pragma warning(default : 4201)
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#ifdef _WIN32
+// Required for Sleep(DWORD)
+#include <Windows.h>
+#else
+// Required for sleep(unsigned int)
+#include <unistd.h>
+#endif
+
+#include <az_iot_hub_client.h>
+#include <az_result.h>
+#include <az_span.h>
+
+// TODO: #564 - Remove the use of the _az_cfh.h header in samples.
+//              Note: this is required to work-around MQTTClient.h as well as az_span init issues.
+#include <_az_cfg.h>
+
+// Device ID
+#define DEVICE_ID "AZ_IOT_DEVICE_ID"
+
+// IoT Hub FQDN
+#define IOT_HUB_FQDN "AZ_IOT_HUB_FQDN"
+
+// Device information
+#define REGISTRATION_ID_ENV "AZ_IOT_REGISTRATION_ID"
+
+// AZ_IOT_DEVICE_X509_CERT_PEM_FILE is the path to a PEM file containing the device certificate and
+// key as well as any intermediate certificates chaining to an uploaded group certificate.
+#define DEVICE_X509_CERT_PEM_FILE "AZ_IOT_DEVICE_X509_CERT_PEM_FILE"
+
+// AZ_IOT_DEVICE_X509_TRUST_PEM_FILE is the path to a PEM file containing the server trusted CA
+// This is usually not needed on Linux or Mac but needs to be set on Windows.
+#define DEVICE_X509_TRUST_PEM_FILE "AZ_IOT_DEVICE_X509_TRUST_PEM_FILE"
+
+#define NUMBER_OF_MESSAGES 5
+
+static char device_id[64];
+static char iot_hub_fqdn[128];
+static char x509_cert_pem_file[512];
+static char x509_trust_pem_file[256];
```

Comment:
>256 [](start = 32, length = 3)

Interesting buffer size for trusted cert ? Did you test cert path?

Reply:
hope I am incorrect.

---
In reply to: [418658585](https://github.com/Azure/azure-sdk-for-c/pull/648#discussion_r418658585) [](ancestors = 418658585)

Reply:
It's a cert path, not a cert.

Reply:
aha .. my bad

---
In reply to: [418697312](https://github.com/Azure/azure-sdk-for-c/pull/648#discussion_r418697312) [](ancestors = 418697312)

Reply:
You were right though: likely my mistake - all paths should have the same value - here's one case where the comment above does make sense - we should have a MAX_PATH define.
This needs to be done to all samples incl. provisioning. 

@danewalton we can do that in a different PR - let me know what you decided.